appcues/src/main/java/com/appcues/ui/EmbedViewManager.kt

```diff
@@ -0,0 +1,90 @@
+package com.appcues.ui
+
+import android.app.Activity
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.ComposeView
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.findViewTreeLifecycleOwner
+import com.appcues.RenderContextManager
+import com.appcues.data.model.RenderContext
+import com.appcues.monitor.AppcuesActivityMonitor
+import com.appcues.monitor.AppcuesActivityMonitor.ActivityMonitorListener
+import com.appcues.ui.composables.AppcuesComposition
+import com.appcues.ui.primitive.EmbedChromeClient
+import com.appcues.ui.utils.getParentView
+import org.koin.core.scope.Scope
+
+internal class EmbedViewManager(
```

Comment:
curious what led to the modal/embed split here versus the previous OverlayViewManager. Looks like some duplication but I'm guessing other things led to too much complication when trying to combine?

Reply:
this was a first pass, I tried to not mess with appcuesOverlay too much, I can come back now and see if there is a better way to structure it.

but one thing I'm thinking here is that maybe we would want to have different classes instead of 1 class with too much responsibility